euphonic/broadening.py

```diff
@@ -0,0 +1,167 @@
+"""
+Functions for broadening spectra
+"""
+from scipy.optimize import nnls
+from scipy.signal import convolve
+import numpy as np
+
+from euphonic import ureg, Quantity
+
+def variable_width(bins: Quantity,
```

Comment:
At the moment I've changed the name of the fast method to `width_interpolated_broadening` but also thought it could be called one of the following: `log_spaced_kernel_broadening`, `width_scaled_spectra_broadening`, or `approximate_weight_scaled_broadening` - not sure how to convey the idea of the method in a more concise way or if these names would even make sense to people not already familiar with the method...